### PR TITLE
Edit page

### PR DESCRIPTION
**Pull request by FNKMYR**
Rather huge PR, so thanks in advance!

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -2,19 +2,22 @@ import { useState } from 'react';
...
 import ExpensesPage from './pages/ExpensesPage.js';
 import AddExpensePage from './pages/AddExpensePage.js';
-import SampleData from './data/sampleData.js';
+import EditExpensePage from './pages/EditExpensePage.js';
 
 export default function App() {
   const [members, setMembers] = useState([]);
-  const [expenses, setExpenses] = useState(SampleData);
+  const [expenses, setExpenses] = useState([]);
+  const [editExpense, setEditExpense] = useState();
```
Comment (macfrei) [inline src/App.js:10]: Usually we want to set an initial state.

File: src/App.js [hunk 2]
```diff
@@ -2,19 +2,22 @@ import { useState } from 'react';
...
   return (
     <Routes>
       <Route
         path="/"
         element={
           <ExpensesPage
             expenses={expenses}
-            setExpenses={setExpenses}
+            setExpenses={value => setExpenses(value)}
+            setEditExpense={value => setEditExpense(value)}
```
Comment (macfrei) [inline src/App.js:20]: Usually when passing callback functions as props we use the prefix `on`.
```suggestion
            onSetExpenses={value => setExpenses(value)}
            onSetEditExpense={value => setEditExpense(value)}
```

File: src/components/Expense.js [hunk 3]
```diff
@@ -10,10 +13,15 @@ export default function Expense({
   paidBy,
   paidFor,
 }) {
+  function handleClick() {
+    setEditExpense(expense);
+    console.log(expense);
```
Comment (macfrei) [inline src/components/Expense.js:18]:
```suggestion
```

File: src/components/Expense.js [hunk 4]
```diff
@@ -10,10 +13,15 @@ export default function Expense({
...
+  }
+
   return (
-    <Wrapper>
+    <Wrapper to="/edit" onClick={handleClick}>
```
Comment (macfrei) [inline src/components/Expense.js:22]: I would recommend against using a `Link` as a wrapper and mixing it with an `onClick` event. I would suggest to use `navigate` from the `useNavigate` (or so) hook from `react-router-dom` to navigate when the function is called.

Also, a `button` with a descriptive text ("Edit Expense") would semantically be more correct.

```js
const navigate = useNavigate()

function handleClick() {
    setEditExpense(expense);
    navigate("/edit")
  }

  return (
    <Wrapper onClick={handleClick}>
```

File: src/components/ExpenseUseForm.js [hunk 5]
```diff
@@ -0,0 +1,458 @@
...
+    console.log(data);
+    console.log(expenses);
+    console.log(data.id);
+    console.log(expenses.findIndex(expense => expense.id === data.id));
+    console.log(
+      expenses.slice(
+        0,
+        expenses.findIndex(expense => expense.id === data.id)
+      )
+    );
```
Comment (macfrei) [inline src/components/ExpenseUseForm.js:41]:
```suggestion
```

File: src/components/ExpenseUseForm.js [hunk 6]
```diff
@@ -0,0 +1,458 @@
...
+          ...expenses.slice(
+            0,
+            expenses.findIndex(expense => expense.id === data.id)
+          ),
+          { ...data },
+          ...expenses.slice(
+            expenses.findIndex(expense => expense.id === data.id) + 1
+          ),
+        ])
+      : // Lösung: setExpenses expeneses slicen vorher, dann element bearbeiten dann slicen und alles spreaden
```
Comment (macfrei) [inline src/components/ExpenseUseForm.js:54]:
```suggestion
      : 
```

File: src/components/ExpenseUseForm.js [hunk 7]
```diff
@@ -0,0 +1,458 @@
...
+          <Error>{errors.amount && <p>{errors.amount.message}</p>}</Error>
+          <Error>{errors.date && <p>{errors.date.message}</p>}</Error>
+          <Error>{errors.paidBy && <p>{errors.paidBy.message}</p>}</Error>
+          <Error>{errors.paidFor && <p>{errors.paidFor.message}</p>}</Error>
+
+          <InputContainer>
+            <StyledInput
+              type="text"
+              id="title"
+              placeholder=" "
```
Comment (macfrei) [inline src/components/ExpenseUseForm.js:77]: If you don't use a placeholder you don't need to set the attribute. For all inputs in this form. 🙂

**Review by macfrei — COMMENTED** [review]
Hi, nice job! I added a few comments you might want to consider 👍 🙂

File: src/components/ExpenseUseForm.js (hunk 7 above)
Comment (FNKMYR) [reply src/components/ExpenseUseForm.js:66]: Actually necessary as I use the 'placeholder-shown' pseudo-class to move the labels as soon as you focus the input. (Line 284)

**Pull request merged**